test/teos/unit/test_chain_monitor.py

```diff
@@ -13,148 +15,245 @@ def test_init(block_processor):
+        watcher_block = chain_monitor.receiving_queues[0].get()
+        responder_block = chain_monitor.receiving_queues[1].get()
         assert watcher_block == responder_block
-        assert chain_monitor.watcher_queue.empty()
-        assert chain_monitor.responder_queue.empty()
+        assert chain_monitor.receiving_queues[0].empty()
+        assert chain_monitor.receiving_queues[1].empty()
 
-    chain_monitor.terminate = True
+    chain_monitor.terminate()
     # The zmq thread needs a block generation to release from the recv method.
     generate_blocks(1)
 
 
+def test_activate_wrong_status_raises(block_processor):
+    # calling activate when not listening should raise
+    chain_monitor = ChainMonitor([Queue(), Queue()], block_processor, bitcoind_feed_params)
+
+    for status in ChainMonitorStatus:
+        if status != ChainMonitorStatus.LISTENING:
+            chain_monitor.status = status  # mock the status
+            with pytest.raises(RuntimeError, match="can only be called in LISTENING status"):
+                chain_monitor.activate()
+
+
 def test_monitor_chain_single_update(block_processor):
     # This test tests that if both threads try to add the same block to the queue, only the first one will make it
-    chain_monitor = ChainMonitor(Queue(), Queue(), block_processor, bitcoind_feed_params)
+    chain_monitor = ChainMonitor([Queue(), Queue()], block_processor, bitcoind_feed_params)
 
     chain_monitor.best_tip = None
     chain_monitor.polling_delta = 2
 
     # We will create a block and wait for the polling thread. Then check the queues to see that the block hash has only
     # been added once.
     chain_monitor.monitor_chain()
+    chain_monitor.activate()
     generate_blocks(1)
 
-    watcher_block = chain_monitor.watcher_queue.get()
-    responder_block = chain_monitor.responder_queue.get()
-    assert watcher_block == responder_block
-    assert chain_monitor.watcher_queue.empty()
-    assert chain_monitor.responder_queue.empty()
+    assert len(chain_monitor.receiving_queues) == 2
+
+    queue0_block = chain_monitor.receiving_queues[0].get()
+    queue1_block = chain_monitor.receiving_queues[1].get()
+    assert queue0_block == queue1_block
+    assert chain_monitor.receiving_queues[0].empty()
+    assert chain_monitor.receiving_queues[1].empty()
 
     # The delta for polling is 2 secs, so let's wait and see
     time.sleep(2)
-    assert chain_monitor.watcher_queue.empty()
-    assert chain_monitor.responder_queue.empty()
+    assert chain_monitor.receiving_queues[0].empty()
+    assert chain_monitor.receiving_queues[1].empty()
 
     # We can also force an update and see that it won't go through
-    assert chain_monitor.update_state(watcher_block) is False
+    assert chain_monitor.enqueue(queue0_block) is False
+
+    chain_monitor.terminate()
+    # The zmq thread needs a block generation to release from the recv method.
+    generate_blocks(1)
+
+
+def test_terminate(block_processor):
```

Comment:
It'll be nice to generate blocks and check that they are not received.

Reply:
Done in dc09ef0.